arangod/Transaction/Methods.cpp

```diff
@@ -1520,7 +1528,8 @@ OperationResult transaction::Methods::insertLocal(
         if (cc != nullptr) {
           // nullptr only happens on controlled shutdown
           size_t nrDone = 0;
-          size_t nrGood = cc->performRequests(requests, chooseTimeout(count),
+          size_t nrGood = cc->performRequests(requests,
+                                              chooseTimeout(count, body->size()*followers->size()),
```

Comment:
I am happy with it as it is, but this one surprised me: the requests to all followers are sent concurrently, so, apart from outgoing network bandwidth (which might be a good argument!), I would expect that the time needed to send off all requests is proportional to `body->size()` rather than `body->size()*followers->size()`. The only reason why I am mentioning this at all is that increasing the timeout here increases the time a failover can happen in case of a failure of a follower (for example in case of a network split). It can only be dropped after the timeout. What do you think?

Reply:
I was torn as to whether body size alone was correct. I chose the conservative solution by saying “no”.  My fear was that the network stack and network infrastructure were shared therefore total size might matter.  But I did not attempt to prove this theory.  I can raise the replication factor today and retest.